src/pages/index.js

```diff
@@ -1,248 +1,250 @@
+const profilePhotoEdit = new PopupWithForm({
+  popupSelector: "#profilePhotoModal",
+  handleFormSubmit: handleProfilePhotoSubmit,
+});
 
-contentAddButton.addEventListener("click", () => {
-  cardFormValidator.toggleButtonState();
-  newCardPopup.openModal();
+const deleteConfirmation = new PopupWithForm({
+  popupSelector: "#deleteContent",
+});
+
+const contentCardPreview = new PopupWithImage({
+  popupSelector: "#jsPopupSelector",
 });
-// contentAddModalClose.addEventListener("click", () => newCardPopup.closeModal());
-
-// ---- //Esc Button Modal Close ---------------------------------------------------------
-
-// function handleEscButton(event) {
-//   const key = event.key;
-//   if (key === "Escape") {
-//     const openedPopup = document.querySelector(".modal_opened");
-//     closeModal(openedPopup);
-//   }
-// }
-// //Overlay Modal Close
-// function handleOverlayClose(evt) {
-//   if (evt.target.classList.contains("modal_opened")) {
-//     closeModal(evt.target);
-//   }
-// }
-
-// const submitButtonEvent = document.querySelector(".modal__form");
-
-// submitButtonEvent.addEventListener("submit", () =>
-//   newUserInfo.setUserInfo({
-//     name: nameInput.value,
-//     about: jobInput.value,
-//   })
-// );
+
+function handleCardDeleteClick(card) {
+  const submitButton = document.querySelector("#confirmDeletion");
+  deleteConfirmation.openModal();
+  let submitButtonPressed = false;
+  submitButton.addEventListener("click", () => {
+    submitButtonPressed = true;
+  });
+  deleteConfirmation.setSubmitAction(() => {
+    if (submitButtonPressed) {
+      api
+        .deleteCard(card.getId())
+        .then(() => {
+          card.remove();
+          deleteConfirmation.closeModal();
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    } else {
+      console.log("delete canceled");
+    }
+    submitButtonPressed = false;
+  });
+}
+
+function handleImageClick(card) {
+  contentCardPreview.openModal(card);
+}
+
+function handleProfilePhotoSubmit(inputValues) {
+  const submitButtonm = document.getElementById("modalPhotoSubmit");
+  const originalButtonText = submitButtonm.textContent;
+  // Change button text to "Saving..."
+  submitButtonm.textContent = "Saving...";
+
+  api
+    .updateProfilePhoto(inputValues)
+    .then((res) => {
+      newUserInfo.setUserInfo(res);
+      profilePhotoEdit.closeModal();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButtonm.textContent = originalButtonText;
+    });
+}
 
 function handleProfileFormSubmit(inputValues) {
-  // profileName.textContent = nameInput.value;
-  // profileJob.textContent = jobInput.value;
-  // newUserInfo.setUserInfo({
-  //   name: inputValues.name,
-  //   about: inputValues.job,
-  // });
-  newUserInfo.setUserInfo(inputValues);
-  profileCardPopup.closeModal();
+  profileFormValidator.toggleButtonState();
```

Comment:
```js

  profileFormValidator.toggleButtonState();
  profileCardPopup.openModal();
```

This is not needed here because it's a submit handler


You don't open the modal here